scipy/stats/_fit.py

```diff
@@ -913,6 +914,18 @@ def goodness_of_fit(dist, data, *, known_params=None, fit_params=None,
     larger number of them exceed the value of statistic for `data`, thereby
     inflating the p-value.
 
+    *Implementation Notes*
+
+    The Filliben statistic is calculated using the theoretical medians of the
+    order statistics of the uniform distribution. Results will be slightly
+    different from those computed using the approximation suggested by
+    Filliben [7]_.
+
+    Computation of the Shapiro-Francia statistic [8]_ for arbitrary
+    distributions is numerically challenging. The expected values of the order
+    statistics are calculated using numerical integration; if the integrator
+    emits a warning or fails, consider using the related Filliben statistic.
```

Comment:
Both your first and second statements seem correct.

In the Filliben statistic, we looked at _medians_ of order statistics rather than means/expected values, so we could transform the medians of uniform distribution order statistics via inverse CDF.

Here, we are looking at means/expected values of order statistics, so knowing the expected values of order statistics of the uniform distribution only helps us with the uniform distribution. That is, because of your second statement, the first statement doesn't do us much good in this PR.